### PR TITLE
Add GUI example app

### PR DESCRIPTION
**Pull request by lynn**
Factor out the "core" of host-example, which is now shared by two C# (CLI and GUI) demos.

Here's a screenshot!

image

Known bugs:

* Clicking Connect, Disconnect, then Connect again crashes the app. (Dispose() throws away too much?)
* It turns out setting `RainwayStream.Permissions` doesn't actually do anything, so the input permission checkboxes don't apply to the current stream.

I gather the ongoing Native SDK refactor might solve one or both of these issues.

### CODE REVIEW

**Comment by bengreenier** [comment]
> (Dispose() throws away too much?)

It's actually the second `Initialize` that is the problem IIRC. [internal link](https://github.com/RainwayApp/rainway-sdk/blob/master/runtimes/native/runtime/src/lib.rs#L48).

File: host-example-core/host-example-core.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <RootNamespace>host_example_core</RootNamespace>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="rainway-sdk" Version="0.3.0" />
```
[the remark this answers is not in the document]
Comment (bengreenier) [reply host-example-core/host-example-core.csproj:11]: If you're going into `main` this is fine, if you change base to `beta` then it should be the `0.4.0-beta.1`, please!

File: host-example-core/HostExample.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        /// incoming connection requests from other peers.
+        /// </summary>
+        public bool AcceptIncoming { get; set; } = true;
+
+        /// <summary>
+        /// Are we currently connected?
+        /// </summary>
+        public bool Connected => runtime != null;
+
+        public void Stop()
```
Comment (bengreenier) [inline host-example-core/HostExample.cs:92]: mega nit: would be good to add a small xmldoc comment for this one too

File: host-example-gui/Form1.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
+using System.Diagnostics;
+
+namespace host_example_gui
+{
+    public partial class Form1 : Form
```
Comment (bengreenier) [inline host-example-gui/Form1.cs:5]: nitpicky again, but `Form1` isn't super descriptive, maybe `RainwayExampleForm` ?

File: host-example-gui/Form1.resx [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <assembly alias="System.Drawing" name="System.Drawing, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
+  <data name="$this.Icon" type="System.Drawing.Icon, System.Drawing" mimetype="application/x-microsoft.net.object.bytearray.base64">
+    <value>
+        AAABAAMAMDAAAAEAIACoJQAANgAAACAgAAABACAAqBAAAN4lAAAQEAAAAQAgAGgEAACGNgAAKAAAADAA
```
Comment (bengreenier) [inline host-example-gui/Form1.resx:63]: this is the base64 encoded Rainway-owned icon, right?

**Review by bengreenier — APPROVED** [review]
A few nitpicky suggestions, but overall this looks good! Nice work @lynn, I think this'll be super helpful for folks that have less experience with the command line, but still want to evaluate Rainway.

File: host-example-gui/Form1.resx (hunk 4 above)
Comment (lynn) [reply host-example-gui/Form1.resx:63]: It looks like it's a blank 48x48 png file. There is a separate .ico file checked in (actually I see there's two, and one is unused)

**Pull request merged**